pkg/ccl/crosscluster/logical/logical_replication_job_test.go

```diff
@@ -807,6 +807,51 @@ func TestLogicalAutoReplan(t *testing.T) {
 	require.Greater(t, len(clientAddresses), 1)
 }
 
+// TestLogicalJobResiliency will follow the following steps:
+// Create two tables and scatter them
+// Set up the initial logical replication job and verify the
+// multiple client addresses are stored in the job progress
+// Kill the node holding table A that's ingesting table B
+// See if another node picks up the stream
+func TestLogicalJobResiliency(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
+
+	skip.UnderRace(t, "multi cluster/node config exhausts hardware")
+
+	clusterArgs := base.TestClusterArgs{
+		ServerArgs: base.TestServerArgs{
+			DefaultTestTenant: base.TestControlsTenantsExplicitly,
+			Knobs: base.TestingKnobs{
+				JobsTestingKnobs: jobs.NewTestingKnobsWithShortIntervals(),
+			},
+		},
+	}
+
+	ctx := context.Background()
+
+	server, s, dbA, dbB := setupLogicalTestServer(t, ctx, clusterArgs, 3)
+	defer server.Stopper().Stop(ctx)
+
+	dbBURL, cleanupB := s.PGUrl(t, serverutils.DBName("b"))
+	defer cleanupB()
+
+	CreateScatteredTable(t, dbA, 2)
+	CreateScatteredTable(t, dbB, 2)
+
+	var jobAID jobspb.JobID
+	dbA.QueryRow(t, "CREATE LOGICAL REPLICATION STREAM FROM TABLE tab ON $1 INTO TABLE tab", dbBURL.String()).Scan(&jobAID)
+
+	now := server.Server(0).Clock().Now()
+	t.Logf("waiting for replication job %d", jobAID)
+	WaitUntilReplicatedTime(t, now, dbA, jobAID)
+
+	progress := jobutils.GetJobProgress(t, dbA, jobAID)
+	addresses := progress.Details.(*jobspb.Progress_LogicalReplication).LogicalReplication.StreamAddresses
+
+	require.Greaterf(t, len(addresses), 1, "The addresses weren't persisted in system.job_info")
```

Comment:
nit: feel free to remove the error message or make it, "less than 2 addresses were persisted"